services/skus/controllers.go

```diff
@@ -1386,15 +1386,15 @@ func handleSubmitReceipt(svc *Service, valid *validator.Validate) handlers.AppHa
 		}
 
 		{
-			exists, err := svc.ExternalIDExists(ctx, extID)
-			if err != nil {
+			_, err := svc.orderRepo.GetByExternalID(ctx, svc.Datastore.RawDB(), extID)
+			if err != nil && !errors.Is(err, model.ErrOrderNotFound) {
 				l.Warn().Err(err).Msg("failed to lookup external id")
 
 				return handlers.WrapError(err, "failed to lookup external id", http.StatusInternalServerError)
 			}
 
-			if exists {
-				return handlers.WrapError(err, "receipt has already been submitted", http.StatusBadRequest)
+			if err == nil {
+				return handlers.WrapError(err, "receipt has already been submitted", http.StatusConflict)
```

Comment:
suggestion(non-blocking): would checking for the order make the intent clearer i.e. if we find an order return opposed to if we dont find an error return.  

imo checking for `err == nil` adds more cognitive load i.e. requires more context such as `err != nil && !errors.Is(err, model.ErrOrderNotFound)` and knowing the behaviour of `GetByExternalID` than:

```
order, err := svc.orderRepo.GetByExternalID(ctx, svc.Datastore.RawDB(), extID)
...
if order != nil {
    return handlers.WrapError(err, "receipt has already been submitted", http.StatusConflict)
}

Reply:
Decided to keep it like that for now.